namer/consul/src/main/scala/io/buoyant/namer/consul/ConsulInitializer.scala

```diff
@@ -42,6 +43,11 @@ case class ConsulConfig(
   override val experimentalRequired = true
 
   @JsonIgnore
+  def interruptionMonitor = Monitor.mk {
+    case e: Failure if e.isFlagged(Failure.Interrupted) => true
+  }
```

Comment:
Would you mind moving this into newNamer so it's clearer how this is used (it's not really part of the config).